### ISSUE TITLE
which label tool should I choose？

### ISSUE THREAD

**Issue by hjm1990818** [dbolya/yolact#151]
Dear Dbolya,

Sorry for bothering you.
I want to segment one dog from an image, I have two questions.
1. which label tool should I choose? LabelImg or Labelme?
2.I just want to detect one category(dog), Should I make any changes to your source code?
Looking forward to hearing from you, thank you very much in advance.

With best regards,
Ming

**Comment by dbolya** [comment]
1. I'm not sure, since I actually haven't used either, but there's already a guide for labelme (see the readme), so probably go with that.
2. There are options. The pretrained weights already have a dog class (since COCO has dogs), so check if that performance is adequate for your needs. If it's not, you can get better performance by fine tuning the classifier on just the dog class, but that will require a little tweaking of both the dataset and the code, so let me know if you want to go that route.

Also, did you mean you want to segement one dog from one image, or just one dog from a bunch of different images of that same dog? YOLACT doesn't do very well with a super small amount of data, so you might not want to train on just that. You can, however, combine the COCO dog instances with your dog instances to get a much larger training set that should be enough.

**Comment by hjm1990818** [comment]
> 1. I'm not sure, since I actually haven't used either, but there's already a guide for labelme (see the readme), so probably go with that.
> 2. There are options. The pretrained weights already have a dog class (since COCO has dogs), so check if that performance is adequate for your needs. If it's not, you can get better performance by fine tuning the classifier on just the dog class, but that will require a little tweaking of both the dataset and the code, so let me know if you want to go that route.
> 
> Also, did you mean you want to segement one dog from one image, or just one dog from a bunch of different images of that same dog? YOLACT doesn't do very well with a super small amount of data, so you might not want to train on just that. You can, however, combine the COCO dog instances with your dog instances to get a much larger training set that should be enough.

Dear Dbolya, I want to segement one dog from one image, what should I do now ? Looking forward to hearing from you.

**Comment by dbolya** [comment]
Have you tried running the current results on that image?

In your yolact directory, create a "weights" folder and download "yolact_base_54_800000.pth" from the readme (the front page of the repo) and put it in the weights folder. Then from the yolact directory, run:
```shell
python eval.py --trained_model=weights/yolact_base_54_800000.pth --score_threshold=0.3 --top_k=100 --image=my_image.png
```
and replace "my_image.png" with the path to your image.

Does it segment out the dog?

**Comment by hjm1990818** [comment]
> Have you tried running the current results on that image?
> 
> In your yolact directory, create a "weights" folder and download "yolact_base_54_800000.pth" from the readme (the front page of the repo) and put it in the weights folder. Then from the yolact directory, run:
> 
> ```shell
> python eval.py --trained_model=weights/yolact_base_54_800000.pth --score_threshold=0.3 --top_k=100 --image=my_image.png
> ```
> 
> and replace "my_image.png" with the path to your image.
> 
> Does it segment out the dog?



Dear Dbolya,

Thank you for your reply.

Yes, it can segment the dog.

Another two questions:  If I want to segment a new object that not been included in COCO dataset, such as a fish. How many images should I prepare for the training set? and  Since I only detect a new category (only fish), what changes should I do on your source code ? 

Looking forward to hearing from you.

My sincerely thanks to you for the time and patience that had been spent in helping me.

Best regards,
Ming

**Comment by dbolya** [comment]
I don't know how many images, since YOLACT is pretty data hungry. I'd say between 100-1000 (though I suggest you find a dataset with fish already segmented, since doing that all yourself would be cumbersome).

The readme (front page of the repo) has all the information you need to create and train on a custom dataset. If you want to add a class, you just follow that procedure and instead of creating an entirely new dataset, you combine the images and annotations of the training set I already use (COCO2017) and your new data. You can also use #36 to fine tune instead of training from scratch.

**Comment by hjm1990818** [comment]
> I don't know how many images, since YOLACT is pretty data hungry. I'd say between 100-1000 (though I suggest you find a dataset with fish already segmented, since doing that all yourself would be cumbersome).
> 
> The readme (front page of the repo) has all the information you need to create and train on a custom dataset. If you want to add a class, you just follow that procedure and instead of creating an entirely new dataset, you combine the images and annotations of the training set I already use (COCO2017) and your new data. You can also use #36 to fine tune instead of training from scratch.

Dear Dbolya,

Thank you very much for your help.

With best regards,
Ming

**Comment by dbolya** [comment]
No problem! I'll close this issue for now. If you need any more help, feel free to reopen the issue or to make a new one.
